docs/gallery/domain/ecephys.py

```diff
@@ -5,152 +5,291 @@
+        description='electrode group for shank {}'.format(ishank),
+        device=device,
+        location='brain area'
+    )
+    # add electrodes to the electrode table
+    for ielec in range(nchannels_per_shank):
+        nwbfile.add_electrode(
+            x=5.3, y=1.5, z=8.5, imp=np.nan,
+            location='unknown',
+            filtering='unknown',
+            group=electrode_group,
+            label='shank{}elec{}'.format(ishank, ielec)
+        )
+        electrode_counter += 1
 
 #######################
-# After setting up electrode group metadata, you should add metadata about the individual electrodes comprising
-# each electrode group. This is done with :py:func:`~pynwb.file.NWBFile.add_electrode`.
-#
-# The first argument to :py:class:`~pynwb.file.NWBFile.add_electrode` is a unique identifier that the user should
-# assign. For details on the rest of the arguments, please see the
-# :py:func:`API documentation <pynwb.file.NWBFile.add_electrode>`.
-
+# Similarly to the ``trials`` table, we can view the ``electrodes`` table in tabular form
+# by converting it to a pandas :py:class:`~pandas.DataFrame`.
 
-for idx in [1, 2, 3, 4]:
-    nwbfile.add_electrode(id=idx,
-                          x=1.0, y=2.0, z=3.0,
-                          imp=float(-idx),
-                          location='CA1', filtering='none',
-                          group=electrode_group)
 
+nwbfile.electrodes.to_dataframe()
 
 #######################
-# .. note: :py:class:`~pynwb.file.NWBFile.add_electrode` returns the index of the electrode you just added.
-#    This can come in handy when creating with an :py:class:`~pynwb.core.DynamicTableRegion`
-#    :py:class:`~pynwb.file.NWBFile.create_electrode_table_region` (See :ref:`ec_recordings`).
+# .. note:: When we added an electrode with the :py:class:`~pynwb.file.NWBFile.add_electrode`
+#    method, we passed in the :py:class:`~pynwb.ecephys.ElectrodeGroup` object for the ``"group"`` argument.
+#    This creates a reference from the ``"electrodes"`` table to the individual :py:class:`~pynwb.ecephys.ElectrodeGroup`
+#    objects, one per row (electrode).
 
 #######################
 # .. _ec_recordings:
 #
 # Extracellular recordings
-# ^^^^^^^^^^^^^^^^^^^^^^^^
+# ------------------------------
+#
+# Raw voltage traces and local-field potential (LFP) data are stored in :py:class:`~pynwb.ecephys.ElectricalSeries`
+# objects. :py:class:`~pynwb.ecephys.ElectricalSeries` is a subclass of :py:class:`~pynwb.base.TimeSeries` specialized for voltage data.
+# To create the :py:class:`~pynwb.ecephys.ElectricalSeries` objects, we need to reference a set of rows
+# in the ``"electrodes"`` table to indicate which electrodes were recorded.
+# We will do this by creating a :py:class:`~pynwb.core.DynamicTableRegion`, which is a type of link that allows you to reference
+# specific rows of a :py:class:`~hdmf.common.table.DynamicTable`, such as the electrodes table, by row indices.
```

Comment:
:py:method:~pynwb.file.NWBFile.create_electrode_table_region` is a convenience function that creates a :py:class:`~pynwb.core.DynamicTableRegion` which references the ``"electrodes"`` table.